Add Developer Preview guidance to the SSG (#239)
redhat-documentation/supplementary-style-guide: supplementary_style_guide/style_guidelines/support.adoc

[[support]]
= Support

[[developer-preview-guidance]]
== Developer Preview

Developer Preview features provide early access to upcoming product features in advance of their possible inclusion in a Red Hat product offering, enabling customers to test functionality and provide feedback during the development process. These features might not have any documentation, are subject to change or removal at any time, and testing is limited. Red Hat might provide ways to submit feedback on Developer Preview features without an associated SLA.

> When documenting a Developer Preview feature, follow these guidelines:

Author: Added something, but LMK if you think we should expand on it.

* Add an admonition labeled IMPORTANT at the beginning of the Developer Preview content and include the template text.
* Use initial uppercase capitalization, that is, Developer Preview.
* Never use the phrase "supported as a Developer Preview", and avoid using "support" in Developer Preview descriptions. Instead, use neutral words like "available", "provide", "capability", and so on.

> * When the Developer Preview feature becomes generally available, remove the IMPORTANT admonition from any document that includes content about the feature.

Reviewer: We might want to say something here about the possibility of it moving into TP status first.

> For more information about the support scope of Red Hat Developer Preview

Reviewer: Just FYI that PR #237 has been merged, whenever you're ready to pull in some of those changes

[[support]]
= Support

[[developer-preview-guidance]]
== Developer Preview

> Developer Preview software provides early access to a technology, component, or feature, in advance of its possible inclusion in a Red Hat product offering. This enables customers to test the functionality and provide feedback during the development process. Documentation is not required for Developer Preview software. Documentation that is provided for Developer Preview software is subject to change or removal at any time, and testing is limited. Red Hat might provide ways to submit feedback on Developer Preview software without an associated SLA.

Reviewer 2: ```suggestion
Developer Preview software provides early access to a technology, component, or feature, in advance of its possible inclusion in a Red Hat product offering. Customers can use Developer Preview software to test functionality and provide feedback during the development process. Documentation is not required for Developer Preview software, but if documentation is provided, it is subject to change or removal at any time. Also, testing is limited for Developer Preview software. Red Hat might provide ways to submit feedback on Developer Preview software without an associated SLA.
```
- Same comment as above re `Red Hat` and non-breaking spaces.
- Same comment as above re anthropomorphism. Also "This" doesn't have a clear antecedent.
- I think this needs better logical flow. Consider my edits as one possible way to improve the flow.

Maybe consider just using the identical wording to the Glossary entry??

Author: I went with your suggested edit thanks.

====

When documenting a Developer Preview software, follow these guidelines:

* Add an admonition labeled **IMPORTANT** at the beginning of the Developer Preview content and include the template text.

* Use initial uppercase capitalization, that is, Developer Preview.

* Never use the phrase "supported as a Developer Preview", and avoid using "support" in Developer Preview descriptions. Instead, use neutral words like "available", "provide", "capability", and so on.

* When the Developer Preview software becomes generally available, remove the IMPORTANT admonition from any document that includes content about the feature.
+
[NOTE]
====
You might need to replace the Developer Preview admonition with a Technology Preview admonition. For more information, see <<Technology Preview>>.
====

Use the following template. Replace _<software_name>_ with the software name:

.Example AsciiDoc: Developer Preview admonition template
[source,text,subs="+quotes"]
----
[IMPORTANT]
====

> _<software_name>_ is Developer Preview software only. Developer Preview software is not supported by Red Hat in any way and is not functionally complete or production-ready. Do not use Developer Preview software for production or business-critical workloads. Developer Preview software provides early access to upcoming product software in advance of its possible inclusion in a Red Hat product offering, enabling customers to test functionality and provide feedback during the development process. This software might not have any documentation, is subject to change or removal at any time, and testing is limited. Red Hat might provide ways to submit feedback on Developer Preview software without an associated SLA.

Reviewer 2: ```suggestion
_<software_name>_ is Developer Preview software only. Developer Preview software is not supported by Red Hat in any way and is not functionally complete or production-ready. Do not use Developer Preview software for production or business-critical workloads. Developer Preview software provides early access to upcoming product software in advance of its possible inclusion in a Red Hat product offering. Customers can use this software to test functionality and provide feedback during the development process. This software might not have any documentation, is subject to change or removal at any time, and has received limited testing. Red Hat might provide ways to submit feedback on Developer Preview software without an associated SLA.
```
- See my similar comments above about anthropomorphism and non-breaking spaces with Red Hat.
- Reword the penultimate sentence so that the items listed in the series are grammatically parallel.

> _<software_name>_ is Developer Preview software only. Developer Preview software is not supported by Red Hat in any way and is not functionally complete or production-ready. Do not use Developer Preview software for production or business-critical workloads. Developer Preview software provides early access to upcoming product software in advance of its possible inclusion in a Red Hat product offering. Customers can use this software to test functionality and provide feedback during the development process. This software might not have any documentation, is subject to change or removal at any time, and has received limited testing. Red Hat might provide ways to submit feedback on Developer Preview software without an associated SLA.

Reviewer 3: Ditto comment above.

Author: See above. Leaving this as is.
